.github/workflows/ci.yml

```diff
@@ -246,6 +252,13 @@ jobs:
 
       - uses: Swatinem/rust-cache@v1
 
+      # This check makes sure the crate dependency check is accurate
+      - name: Check Cargo.lock is up to date
+        uses: actions-rs/cargo@v1.0.3
+        with:
+          command: check
+          args: --locked --all-features --all-targets
```

Comment:
I think this change fixes #3388.

The previous CI config ran clippy (which updated Cargo.lock) before checking if Cargo.lock was up to date.

Reply:
That's a possibility. I moved it to the `build` job before starting the build, it seemed like a better place. As this was located after a Clippy action that changed the environment to run Clippy, it could have been affecting the output of this step.